bredis/AsyncConnection.hpp

```diff
@@ -51,24 +53,75 @@ template <typename S> class AsyncConnection {
     std::mutex tx_queue_mutex_;
     std::mutex rx_queue_mutex_;
 
+    // I think that queuing should be the responsibility
+    // of a higher level interface and not this class.
+    // Instead, this class should just allow for one
+    // active asynchronous Redis read and one active
+    // asynchronous Redis write, with undefined behavior
+    // if two concurrent reads or two concurrent writes
+    // happen.
+    //
+    // Rationale: TCP/IP provides application-level flow
+    // control to match the rate of the receiver and the
+    // sender. In this implementation, the reader attempts
+    // to read as quickly as possible no matter how fast
+    // the consumer (in this case the command_callback_t)
+    // can process the data. This not only circumvents
+    // TCP/IP's natural application level flow control,
+    // but allows the size of the respective queues to
+    // grow without bounds. Furthermore, the caller is
+    // given no opportunity to provide "back pressure",
+    // nor is the caller able to detect the unbounded
+    // growth and impose its own policies on it.
+    // 
```

Comment:
After a few mediataions, I have came to the solution that the plurality of single command / multiple commands shuld be  pushed out of `AsyncConnection` into `command_wrapper_t`:


https://github.com/basiliscos/cpp-bredis/blob/try-avoid-queuing/include/bredis/Command.hpp

https://github.com/basiliscos/cpp-bredis/blob/try-avoid-queuing/include/bredis/AsyncConnection.hpp#L57

which is type-erased boost-variant for singe command / multiple commands. Then it became possible to one large buffer to `write`. The performance became comparable 

```
./t-11-multi-ping  0.61s user 0.06s system 11% cpu 5.692 total
```

vs old:

```
./t-11-multi-ping  0.27s user 0.01s system 87% cpu 0.322 total
```

Actually I think this difference id is due to the fact, that in `v0.01` callbacks are not allocated during main proccessing, meanwhile new `read_callback` is everytime the previous one is finished.

That also can be managed (or even improved), if the number of expected replies will be specified on `async_read`. Behind the useless synthetic tests, it might be very useful for `multi/exec` transactions in redis ( https://redis.io/topics/transactions ), where reply for single queued command will be useless.

what do you think?